File: src/templates/express/src/application/services/UserService.js
const UserRepository = require('../../domain/repositories/UserRepository');

class UserService {
  static async getUser(id) {
    return await UserRepository.findById(id);
  }

  static async createUser(userData) {
    return await UserRepository.create(userData);
  }
}

module.exports = UserService;

File: src/templates/rust/src/application/user_service.rs
use crate::domain::user::User;
use crate::infrastructure::repository::UserRepository;
use uuid::Uuid;

pub struct UserService<T: UserRepository> {
    repository: T,
}

impl<T: UserRepository> UserService<T> {
    pub fn new(repository: T) -> Self {
        Self { repository }
    }

    pub async fn list_users(&self) -> Vec<User> {
        self.repository.get_all_users().await
    }

    pub async fn create_user(&self, name: String, email: String) -> User {
        let user = User {
            id: Uuid::new_v4(),
            name,
            email,
        };
        self.repository.add_user(user.clone()).await;
        user
    }
}
